plugins/woocommerce/src/StoreApi/Utilities/QuantityLimits.php

```diff
@@ -73,6 +75,38 @@ public function get_add_to_cart_limits( \WC_Product $product ) {
 		];
 	}
 
+	/**
+	 * Fix a quantity violation by adjusting it to the nearest valid quantity.
+	 *
+	 * @param int   $quantity The quantity to fix.
+	 * @param array $cart_item The cart item.
+	 * @return int
+	 */
+	public function normalize_cart_item_quantity( int $quantity, array $cart_item ) {
+		$product = $cart_item['data'];
```

Comment:
Is it possible for `$cart_item['data']` not to be set? Should we check before accessing it?

Reply:
I think everywhere in core assumes the cart returns valid types. We can add an extra check, it won't hurt.

Reply:
https://github.com/woocommerce/woocommerce/pull/52154/commits/aae25d6c0fdf1aafa2d1ee02fc90f99ff102c1b1